extensions/wikia/ARecoveryEngine/templates/ARecoveryEngineApiController_getPageFairBootstrap.mustache

```diff
@@ -1,13 +1,10 @@
 <!-- Begin PageFair recovery -->
 <script type="text/javascript">
-	require([
-		"ext.wikia.adEngine.adContext"
-	], function (adContext) {
-		var context = adContext.getContext();
-
-		if (context.opts.pageFairRecovery === true && !window.wgUserName) {
-			{{{code}}}
-		}
-	});
+	if (!window.wgUserName &&
+			window.ads &&
+			window.ads.context &&
+			window.ads.context.opts.pageFairRecovery === true) {
+		{{{code}}}
+	}
```

Comment:
IMO it does not solve the issue, it hides the race condition. Instead of throwing an error console it won't enable page fair recovery which will be harder to notice. What do you think about switching those two lines:
https://github.com/Wikia/app/blob/dev/skins/oasis/modules/templates/Oasis_Index.php#L37
(with line 38)